tests/PHPStanTests/nsrt/preg-match.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+namespace PregMatchShapes;
+
+use Composer\Pcre\Preg;
+use function PHPStan\Testing\assertType;
+
+function doMatch(string $s): void
+{
+    if (Preg::match('/Price: /i', $s, $matches)) {
+        assertType('array{string}', $matches);
+    } else {
+        assertType('array{}', $matches);
+    }
+    assertType('array{}|array{string}', $matches);
+
+    if (Preg::match('/Price: (£|€)\d+/', $s, $matches)) {
+        assertType('array{string, string|null}', $matches);
+    } else {
+        assertType('array{}', $matches);
+    }
+    assertType('array{}|array{string, string|null}', $matches);
+
+    if (Preg::match('/Price: (£|€)?\d+/', $s, $matches)) {
+        assertType('array{0: string, 1?: string|null}', $matches);
+    } else {
+        assertType('array{}', $matches);
+    }
+    assertType('array{}|array{0: string, 1?: string|null}', $matches);
+}
+
+function identicalMatch(string $s): void
+{
+    if (Preg::match('/Price: /i', $s, $matches) === 1) {
+        assertType('array{string}', $matches);
+    } else {
+        assertType('array{}', $matches);
+    }
+    assertType('array{}|array{string}', $matches);
+}
+
+function equalMatch(string $s): void
+{
+    if (Preg::match('/Price: /i', $s, $matches) == 1) {
+        assertType('array{string}', $matches);
+    } else {
+        assertType('array{}', $matches);
+    }
+    assertType('array{}|array{string}', $matches);
+}
```

Comment:
we need to drop these tests as these cannot be supported atm.

I tried tackling support for it with https://github.com/phpstan/phpstan-src/pull/3185 but it has some open problems and I think we will not finish it in the near future

Reply:
Ok I can do that myself thanks.

And any clue about the phpstan runtime failure?

> Error: Internal error: Composer\Pcre\PHPStan\PregMatchFlags::getType(): Argument #1 ($flagsArg) must be of type ?Composer\Pcre\PHPStan\Arg, PhpParser\Node\Arg given, called in /home/runner/work/pcre/pcre/src/PHPStan/PregMatchTypeSpecifyingExtension.php on line 62 while analysing file /home/runner/work/pcre/pcre/src/Regex.php

Reply:
Oh now looking at that again it looks like a missing use statement..